# Probability Models


In [1]:
### headers
import numpy as np
from numpy import random

import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

```{contents}
:local:
:depth: 2
```


This section introduces probability language and the table structure used to describe data. The goal is to separate the population or process we care about from the sample we observe, then identify variables, observations, and distributions.


## Probability, Data, and Variables

When describing data, make sure you cover center, spread, and shape.


In [2]:
tips = sns.load_dataset('tips')
tips                      ### size means number of ppl in the party

,total_bill,tip,sex,smoker,day,time,size
0,16.99,1.01,Female,No,Sun,Dinner,2
1,10.34,1.66,Male,No,Sun,Dinner,3
2,21.01,3.50,Male,No,Sun,Dinner,3
3,23.68,3.31,Male,No,Sun,Dinner,2
4,24.59,3.61,Female,No,Sun,Dinner,4
...,...,...,...,...,...,...,...
239,29.03,5.92,Male,No,Sat,Dinner,3
240,27.18,2.00,Female,Yes,Sat,Dinner,2
241,22.67,2.00,Male,Yes,Sat,Dinner,2
242,17.82,1.75,Male,No,Sat,Dinner,2


In [3]:
tips.shape       ### a dataframe has shape, which is a tuple

(244, 7)

### Population vs Sample
Population is the full group we care about (e.g., “all lunch bills at the restaurant this year”), whereas sample is the subset we actually measured (e.g., the 244 rows in the tips dataset). We only ever see the sample, but we talk like we know the population.
  
### Variable vs Observation
Each **row** in the dataframe/dataset is an observation (e.g., one restaurant bill), whereas **columns** are variables (e.g., total_bill, tip, smoker, time (Dinner/Lunch), etc.).

- **Quantitative** variable: e.g., total_bill (numeric, measured)
- **Categorical** variable: time (Lunch vs Dinner), smoker (Yes/No)

### Distribution

A **distribution** shows how often different values occur. For numeric data:
- Are most bills around \$10? \$20? \$50?
- Are there a few very large bills? (outliers)

For categorical data:
- How common is “Dinner” vs “Lunch”?



## Section Practice

Use this short exercise to check the main workflow from this section.


In [4]:
### Exercise: Compute Evidence for Probability Models
# Use the sample values below to summarize a small business metric.
# Calculate the mean and the distance between the largest and smallest values.
# Print both values.
values = [18, 21, 19, 25, 22]
### Your code starts here




### Your code ends here


In [5]:
### Solution
values = [18, 21, 19, 25, 22]
mean_value = sum(values) / len(values)
spread = max(values) - min(values)
print("Mean:", mean_value)
print("Spread:", spread)


Mean: 21.0
Spread: 7
